shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/Generator/Utxo.hs

```diff
@@ -346,21 +475,22 @@ genTxBody inputs outputs certs wdrls update fee slotWithTTL mdHash = do
 --
 -- The idea is to have an specified spending balance and fees that must be paid
 -- by the selected addresses.
+-- TODO need right splitting of v!
 calcOutputsFromBalance ::
-  (HasCallStack, Era era) =>
-  Coin ->
+  (Era era,Val v) =>
```

Comment:
I guess in theory it's nice that this function is `Val`-polymorphic, but in reality it's not - we project down to coin to construct the `TxOut`s. It's only ever called with `Coin`, and the cost of this change is to introduce regressions elsewhere - moving test code back into the main library, adding things back into `NonTraceProperties` etc. So better to leave it fixed at `Coin` for now, and let it be made appropriate polymorphic at the same time as the rest of the testing code.